lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -178,6 +213,59 @@ public void testLoadStats()
         assertEquals(value, "abc");
     }
 
+    @Test(timeOut = TEST_TIMEOUT_MILLIS, invocationCount = 10, successPercentage = 50)
+    public void testLoadFailure()
+            throws Exception
+    {
+        Cache<Integer, String> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(0)
+                .expireAfterWrite(0, DAYS)
+                .shareResultsAndFailuresEvenIfDisabled()
+                .build();
+        int key = 10;
+
+        ExecutorService executor = newFixedThreadPool(2);
+        try {
+            AtomicBoolean first = new AtomicBoolean(true);
+            CyclicBarrier barrier = new CyclicBarrier(2);
+
+            List<Future<String>> futures = new ArrayList<>();
+            for (int i = 0; i < 2; i++) {
+                futures.add(executor.submit(() -> {
+                    barrier.await(10, SECONDS);
+                    return cache.get(key, () -> {
+                        if (first.compareAndSet(true, false)) {
+                            // first
+                            Thread.sleep(1); // increase chances that second thread calls cache.get before we return
+                            throw new RuntimeException("first attempt is poised to fail");
+                        }
+                        return "success";
+                    });
+                }));
+            }
+
+            List<String> results = new ArrayList<>();
+            for (Future<String> future : futures) {
+                try {
+                    results.add(future.get());
+                }
+                catch (ExecutionException e) {
+                    results.add(e.getCause().toString());
+                }
+            }
+
+            // Note: if this starts to fail, that suggests that Guava implementation changed and NoopCache may be redundant now.
+            assertThat(results)
+                    .containsExactlyInAnyOrder(
```

Comment:
nit: those are the same, so constraint relaxation via "inAnyOrder" can be skipped.